### PR TITLE
Split dev packages

### PR DESCRIPTION
**Pull request by at-wat**
- https://github.com/alpine-ros/ros-abuild-docker/issues/24
- requires https://github.com/seqsense/aports-ros-experimental/pull/970 to pass the tests
- docker images: https://github.com/alpine-ros/alpine-ros/pull/39

Automatically split dev dependencies and files to `-dev` subpackage.

All dev packages will be automatically installed if ros-dev meta package is installed.
We can install ros-dev to the base image by default to keep the default behavior.
Users can install packages without dev files by
`apk add !ros-dev ...`

### Dependencies

On master branch, `ros-noetic-rospack` had following deps
```
boost-dev
pkgconf
py3-catkin-pkg
py3-rosdep
python3-dev
ros-noetic-ros-environment
so:libboost_filesystem.so.1.84.0
so:libboost_program_options.so.1.84.0
so:libc.musl-x86_64.so.1
so:libgcc_s.so.1
so:libpython3.12.so.1.0
so:libstdc++.so.6
so:libtinyxml2.so.10
tinyxml2-dev
```

With `--split-dev` option, `ros-noetic-rospack` has
```
pkgconf
py3-catkin-pkg
py3-rosdep
ros-noetic-ros-environment
so:libboost_filesystem.so.1.84.0
so:libboost_program_options.so.1.84.0
so:libc.musl-x86_64.so.1
so:libgcc_s.so.1
so:libpython3.12.so.1.0
so:libstdc++.so.6
so:libtinyxml2.so.10
```
and `ros-noetic-rospack-dev` has
```
boost-dev
pkgconf
python3-dev
tinyxml2-dev
```

With `--split-dev` option, ROS packages in `build_depends` will be resolved as `ros-noetic-package` + `ros-noetic-package-dev` in APKBUILD.

### Files

For example, `ros-noetic-rostime` will contain
```
└── usr
    └── ros
        └── noetic
            ├── lib
            │   └── librostime.so
            └── share
                └── rostime
                    └── package.xml
```
and `ros-noetic-rostime-dev` will contain
```
└── usr
    └── ros
        └── noetic
            ├── include
            │   └── ros
            │       ├── duration.h
            │       ├── impl
            │       │   ├── duration.h
            │       │   └── time.h
            │       ├── rate.h
            │       ├── rostime_decl.h
            │       └── time.h
            ├── lib
            │   └── pkgconfig
            │       └── rostime.pc
            └── share
                └── rostime
                    └── cmake
                        ├── rostimeConfig.cmake
                        └── rostimeConfig-version.cmake
```

### CODE REVIEW

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -11,11 +11,13 @@ jobs:
   build:
     runs-on: ubuntu-latest
     strategy:
+      fail-fast: false
       matrix:
         env:
           - ALPINE_VERSION=3.17 ROS_DISTRO=noetic ROS_DISTRIBUTION_TYPE=ros1
           - ALPINE_VERSION=3.17 ROS_DISTRO=humble ROS_DISTRIBUTION_TYPE=ros2
           - ALPINE_VERSION=3.20 ROS_DISTRO=noetic ROS_DISTRIBUTION_TYPE=ros1
+          - ALPINE_VERSION=3.20 ROS_DISTRO=noetic ROS_DISTRIBUTION_TYPE=ros1 SPLIT_DEV=yes
```
Comment (f-fl0) [inline .github/workflows/build.yaml:20]: Will we keep both alpine 3.20 noetic jobs or was it just for development?

**Review by f-fl0 — APPROVED** [review]
LGTM. Just left a question.

**Review by Taka-Kazu — APPROVED** [review]

File: .github/workflows/build.yaml (hunk 1 above)
Comment (at-wat) [reply .github/workflows/build.yaml:20]: Our Alpine ROS repository will split dev packages, but users still can keep them not splitted when building their packages using ros-abuild-docker image

Comment (f-fl0) [reply .github/workflows/build.yaml:20]: The `ros-abuild:3.20-noetic` image is pushed twice but I guess it is not a problem.

Comment (at-wat) [reply .github/workflows/build.yaml:20]: I'm gonna revert the matrix and test split-dev mode in each matrixed job

Comment (at-wat) [reply .github/workflows/build.yaml]: Fixed CI

Comment (f-fl0) [reply .github/workflows/build.yaml]: Thank you!

Comment (at-wat) [reply .github/workflows/build.yaml]: > Our Alpine ROS repository will split dev packages, but users still can keep them not splitted when building their packages using ros-abuild-docker image

I noticed this is wrong.
As `ros-noetic-catkin` doesn't contain dev files, catkin packages on v3.20 must be always built by split-dev mode.

Comment (at-wat) [reply .github/workflows/build.yaml]: Fixed to test only split-dev mode on v3.20

Comment (at-wat) [reply .github/workflows/build.yaml]: Updated docker image to automatically enable split-dev mode on 3.20

**Pull request merged**